lib/bundler/cli/install.rb

```diff
@@ -182,8 +182,13 @@ def normalize_settings
 
       Bundler.settings.set_command_option_if_given :clean, options["clean"]
 
-      Bundler.settings.set_command_option :without, options[:without] unless Bundler.settings[:without] == options[:without]
-      Bundler.settings.set_command_option :with,    options[:with]    unless Bundler.settings[:with]    == options[:with]
+      unless Bundler.settings[:without] == options[:without] && Bundler.settings[:with] == options[:with]
```

Comment:
It keeps the 1.0 tests passing, because 1.0 has super weird behavior around setting groups